src/main/java/br/com/caelum/mamute/user/api/UserFilterResource.java

```diff
@@ -0,0 +1,61 @@
+package br.com.caelum.mamute.user.api;
+
+import br.com.caelum.mamute.user.domain.SearchCriteria;
+import br.com.caelum.mamute.user.domain.UserEntity;
+import lombok.*;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.data.jpa.domain.Specification;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import java.util.List;

```

Comment:
Opa mano, gostei da ideia desta classe, só fiquei com dúvida em termos uma classe na "camada" de apresentação com dependências de persistência.

Reply:
Mano, essa é uma que queria ver em conjunto uma melhor forma de fazermos isso.
Acho o melhor lugar criar no repositorio msm. Porem como estamos utilizando interface para o repositorio, nao vamos conseguir montar la esses Predicates.

Reply:
Podemos testar com a combinação do private methods + default method do java 9/11.
Vamos ver isso mais p frente :)

Reply:
sim, a principio eu tinha pensado, no default method e fazer a implementacao inteira la, porem iriamos conseguir chamar os metodos de find por la.
Mas pensando melhor posso criar o default method que retorne apenas os Predicates e no service chamo o find.

vou providenciar isso na sequencia, fiquei imcomodado com isso rs.

Reply:
mano, acho que isso pode nos ajudar

https://blog.tratif.com/2017/11/23/effective-restful-search-api-in-spring/